tests/generic/test_fileHDFio.py

```diff
@@ -45,43 +46,79 @@ def tearDownClass(cls):
-                ).flatten()
+            self.assertIsInstance(array, np.ndarray)
+            self.assertEqual(array.dtype, np.dtype('int'))
+
+        with self.subTest('content/array_3d'):
+            array = hdf["content"]["array_3d"]
+            self.assertTrue(
+                all(
+                    np.equal(
+                        array,
+                        np.array([[1, 2, 3], [4, 5, 6]]),
+                    ).flatten()
+                )
             )
-        )
-        self.assertTrue(
-            all(
-                np.equal(
-                    hdf["content/traj"][1], np.array([[7, 8, 9]])
-                ).flatten()
+            self.assertIsInstance(array, np.ndarray)
+            self.assertEqual(array.dtype, np.dtype('int'))
+
+        with self.subTest('content/indices'):
+            array = hdf['content/indices']
+            self.assertTrue(
+                all(
+                    np.equal(
+                        array,
+                        np.array([1, 1, 1, 1, 6])
+                    ).flatten()
+                )
             )
-        )
-        self.assertEqual(hdf["content/dict"]["key_1"], 1)
-        self.assertEqual(hdf["content/dict"]["key_2"], "hallo")
-        self.assertEqual(hdf["content/dict_numpy"]["key_1"], 1)
-        self.assertTrue(
-            all(
-                np.equal(
-                    hdf["content/dict_numpy"]["key_2"],
-                    np.array([1, 2, 3, 4, 5, 6]),
+            self.assertIsInstance(array, np.ndarray)
+            self.assertEqual(array.dtype, np.dtype('int'))
+
+        with self.subTest('content/traj'):
+            array = hdf["content/traj"]
+            self.assertTrue(
+                all(
+                    np.equal(
+                        array[0], np.array([[1, 2, 3], [4, 5, 6]])
+                    ).flatten()
                 )
             )
-        )
-        self.assertEqual(hdf['content/group/some_entry'], 'present')
+            self.assertTrue(
+                all(
+                    np.equal(
+                        array[1], np.array([[7, 8, 9]])
+                    ).flatten()
+                )
+            )
+            self.assertIsInstance(array, np.ndarray)
+            self.assertEqual(array.dtype, np.dtype('O'))
+
+        with self.subTest('content/dict'):
+            content_dict = hdf["content/dict"]
+            self.assertEqual(content_dict["key_1"], 1)
+            self.assertEqual(content_dict["key_2"], "hallo")
+            self.assertIsInstance(content_dict, dict)
+
+        with self.subTest('content/dict_numpy'):
+            content_dict = hdf["content/dict_numpy"]
+            self.assertEqual(content_dict["key_1"], 1)
+            self.assertTrue(
+                all(
+                    np.equal(
+                        content_dict["key_2"],
+                        np.array([1, 2, 3, 4, 5, 6]),
+                    )
+                )
```

Comment:
```suggestion
                np.array_equal(
                     content_dict["key_2"],
                     np.array([1, 2, 3, 4, 5, 6]),
                )
```

Reply:
That pattern comes up a few times, but `array_equal` should be correct for all of them.